airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/uploader/BigQueryUploaderFactory.java

```diff
@@ -141,7 +163,13 @@ private static BigQueryDirectUploader getBigQueryDirectUploader(
         .setProject(bigQuery.getOptions().getProjectId())
         .build();
 
-    final TableDataWriteChannel writer = bigQuery.writer(job, writeChannelConfiguration);
+    final TableDataWriteChannel writer;
+
+    try{
+      writer = bigQuery.writer(job, writeChannelConfiguration);
+    }catch (final BigQueryException e){
+      throw new ConfigErrorException(CONFIG_ERROR_MSG + e);
```

Comment:
is there a way to make this `catch` clause more specific, so that only permission and 404 errors result in `ConfigErrorException`, but other problems (whatever they may be) are not translated into `ConfigErrorException`?

Reply:
To tell the truth, this is what I did from the very beginning - i.e. added a catch to check the only exceptions with 404 in message. But after playing a bit more I noticed that all other possible issues were also related to config issues. So then updated code to catch everything as ConfigError. At this part of code, we just create a new connection. So whatever will go wrong - will be more or less related to the config issue.  Should I anyway update it to catch the 404 only? Thanks

Reply:
Ideally, we'd catch specific error codes and provide error messages specific to those codes. Here is the list of error codes (https://cloud.google.com/bigquery/docs/error-messages). I don't think we can run into all of these here. We can probably assume that we will mostly run into `404` and `403` which we should report as config errors. Occasionally, we might run into `500` and `503`, which we should not report as config errors and we should provide a different error message when we run into `500` and `503`